### DISCUSSION TITLE
Form Editor, day 1

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#264] [General]
I thought some of you might like to see some early progress on the UI side of things...

![tb_formEditor1](https://user-images.githubusercontent.com/25738109/124491153-dc971600-ddaa-11eb-981c-b95f4184b609.gif)

**Comment by camomille22003** [comment]
Have you seen this:

https://youtu.be/RJmDeGAroR4

https://www.vbforums.com/showthread.php?892266-VB-NET-forms-and-controls-for-VBA-VB6

**Comment by uwbwsvd** [comment]
Do you invent the wheel new or is there a blueprint for all this GUI stuff?

**Comment by WaynePhillipsEA** [comment]
@camomille22003 thanks, yes, I've seen that.  However, we're not using any .NET in the backbone of this project given the aim to provide cross-platform support in the not too distant future.  Additionally we need a VS-code integrated form designer.  It's just not suitable unfortunately.

More implementation detail to follow in the coming week.   I'm designing all this in a way that you guys can get really involved if you so wish, so stay tuned.

**Comment by DavidHollman** [comment]
How will the equivalent of the data from FRX files be stored? It always seemed to me that having it in a text-based format would be an improvement, if possible. This would allow more effective source control, diffing, etc. and maybe even hand-editing in a pinch.

**Reply by WaynePhillipsEA** [reply]
JSON :)

The form designer is built as a webview (html + javascript), so it made sense to use JSON.
